### PR TITLE
Login validation

### PR DESCRIPTION
**Pull request by adam-adair**
It's boring but I did it anyway.
--login form error feedback
--new user form error feedback
--bad single art id error
--not found page

### CODE REVIEW

File: client/components/LoginScreen.js [hunk 1]
```diff
@@ -50,7 +74,9 @@ class LoginScreen extends Component {
           <h2>Log in</h2>
           <hr />
           <input name="email" placeholder="Email" onChange={this.handleChange}/>
+          { emError ? <h5 className="noQty">{emError}</h5> : <div /> }
           <input name="password" type="password" placeholder="Password" onChange={this.handleChange}/>
+          { pwError ? <h5 className="noQty">{pwError}</h5> : <div /> }
```
Comment (zoegreene) [inline client/components/LoginScreen.js:81]: Not a bug or anything, but it'd be nice if you cleared out the old error in the emError or pwError section when you go to try again and get a different error.

File: client/components/NotFound.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { Link } from 'react-router-dom'
+
+const NotFound = () => {
+  return(
+    <div className="adminNav"><b>How did I get here? This is not my beautiful house.</b> <Link to="/">Click to go back home.</Link></div>
```
Comment (zoegreene) [inline client/components/NotFound.js:6]: good one chortle chortle

File: client/components/SingleArtwork.js [hunk 3]
```diff
@@ -20,11 +23,14 @@ export class SingleArtwork extends React.Component {
...
 
-    componentDidMount(){
-        this.props.getArtwork(this.props.match.params.id)
+    async componentDidMount(){
+        await this.props.getArtwork(this.props.match.params.id)
+        this.setState({...this.state, loading: false})
     }
 
     render(){
+        if(!this.state.loading && !this.props.artwork.id) return (<h4 className="noQty">This ain't no artwork. This ain't no disco.</h4>)
```
Comment (zoegreene) [inline client/components/SingleArtwork.js]: somebody's having fun

File: client/redux/artwork.js [hunk 4]
```diff
@@ -20,7 +20,7 @@ export const getArtwork = (id) => {
   };
 
 
-export default function artworkReducer( state = [], action) {
+export default function artworkReducer( state = {} , action) {
```
Comment (zoegreene) [inline client/redux/artwork.js]: Thanks for changing! I think I left a comment on this a while back but never checked on it

File: server/api/users.js [hunk 5]
```diff
@@ -68,13 +68,21 @@ router.post('/', async (req, res) => {
...
+      res.sendStatus(400)
     }
   } catch (e) {
-    console.log(e)
-    //do something later to distinguish between existing user and other failure
-    res.status(400).send({
-      message: 'New user creation failed.'
-    })
+    //this checks the type of error coming from sequelize
+    if(e.errors[0].type === 'unique violation') {
```
Comment (zoegreene) [inline server/api/users.js:76]: I like how you've implemented this to be the property of the state in the front end... clever

**Review by zoegreene — APPROVED** [review]
Overall, looks good to me. Left a minor "nice to have" comment about the login screen.

**Review by z-rodney — COMMENTED** [review]
looks good! i like Zoe's point that it would be nice to clear out the error messages that no longer apply

**Comment by adam-adair** [comment]
I resolved merge conflicts w dev and made small update to clear old errors in login form.

**Pull request merged**